blogs/views.py

```diff
@@ -43,4 +38,126 @@ def postBlog(request):
     print(serializer)
     if serializer.is_valid():
         serializer.save()
-        return Response(serializer.data)
\ No newline at end of file
+        return Response(serializer.data)
+    
+
+@api_view(['GET'])
+def paginating(request, page_number):
+    per_page = 10
+    posts = Post.objects.all()[page_number*per_page : ((page_number*per_page)+per_page)]
+    serializer= PostSerializer(posts, many = True)
+    print(serializer)
+    return Response(serializer.data)    
+
+
+@api_view(['GET'])
+def paginating_offset(request, limit, offset_row):
+    # limit = 10
+    # offset_row = 7
+    posts = Post.objects.all()[offset_row:(offset_row+limit)]
+    serializer= PostSerializer(posts, many = True)
+    print(serializer)
+    return Response(serializer.data)    
+
+
+# def pagination_search(request, search):
+#     # if request.method == 'GET':
+#         posts = Post.objects.all()
+       
+        
+#         l = []
+#         for post in posts:
+#             if search in post.title:
+#                 search_post = post.title
+#                 l.append(search_post)
+#                 print(l)
+
+
+#         return JsonResponse(l)    
+
+    # {jtu:80}
+    # str = "{"+"key"+":"+"value"+"}"
+
+# def pagination_search(request, search):
+#         posts = Post.objects.filter(title__icontains = search)
+        
+#         titles = []
+#         for post in posts:
+#             title = post.title
+#             titles.append(title)
+#         print(titles)
+#         return JsonResponse(titles, safe = False)    
+
+
+def pagination_search(request, search, page_number, page_size):
+        posts = Post.objects.filter(title__icontains = search)
+        post_pag = posts[page_number*page_size:page_number*page_size+page_number]
+        print(post_pag)    
+        titles = post_pag.values_list('title', flat=True)
+        print(titles)
+        return JsonResponse(list(titles), safe = False)
+
+
+class global_pag():
+    page_number = 2
+    page_size = 3
+
+    
+    def __init__(self, page_number, page_size):     
+        self.page_number = page_number
+        self.page_size = page_size 
+        
+        # return JsonResponse(list(data), safe = False )
+    
+
+    def all(self):
+        data = Post.objects.all()[self.page_number*self.page_size:self.page_number*self.page_size+self.page_number]  
```

Comment:
these functions should not be here. 
the goal of global pagination is that you can use that class and your page view would get paginated in proper format.